middleware/app.js

```diff
@@ -34,41 +36,30 @@ const checkJwt = jwt({
 app.use(express.json());
 app.use(cors());
 
-// List all vehicles from blockchain
-app.get('/vehicles', async (_, res) => {
+// Retrieve a list of claims from the blockchain.
+app.get('/claims', async (_, res) => {
   try {
-    const VehicleClient = require('./lib/vehicle-client.js');
-    const client = new VehicleClient();
-
-    const resVehicleList = await client.listVehicles();
-    console.log(resVehicleList);
-    res.send(resVehicleList);
+    const client = new ClaimClient();
+    const claimsList = await client.listClaims();
+    console.log(claimsList);
+    res.send(claimsList);
   } catch (err) {
     console.log(err);
     res.sendStatus(500);
   }
 });
 
-// Post a new vehicle to the blockchain
-app.post('/vehicles', checkJwt, async (req, res) => {
+// Post a new claim to the blockchain.
+app.post('/claims', checkJwt, async (req, res) => {
   try {
     const privateKey = req.headers.private_key;
     if (typeof privateKey !== 'string') {
       throw new Error('Invalid request');
     }
-    const VehicleClient = require('./lib/vehicle-client.js');
-    const client = new VehicleClient(privateKey);
-
-    const newVehicle = {
-      vin: req.body.vin,
-      color: req.body.color,
-      model: req.body.model,
-      status: req.body.status,
-    };
-
-    const resPostVehicle = await client.createVehicle(newVehicle);
-    console.log(resPostVehicle);
-    res.send(resPostVehicle);
+    const client = new ClaimClient(privateKey);
+    const response = await client.createClaim(req.body);
```

Comment:
Is this possible because of your protobuf changes, or just something smarter I could have done without having to list each value of the body?

Reply:
Sort of both. 

The way you had it, although ensuring only `vin`, `color`, `model`, and `status` were included in the `createVehicle` call, didn't guarantee that those values existed. So, worst case scenario would have been an object that looked like this...

```js
{
  vin: undefined,
  color: undefined,
  model: undefined,
  status: undefined,
}
```

Because protobufs have strict types, the transaction will simply fail if the caller of the API passes in an object that has properties that aren't defined. Also, if the caller passes an object that is missing properties, those missing properties receive default empty values of that type (so `string` gets `""`, `number` gets `0`, an array of any kind gets `[]`, etc).